staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_status_test.go

```diff
@@ -90,6 +91,58 @@ func TestRolloutStatus(t *testing.T) {
 	}
 }
 
+func TestRolloutStatusNoResources(t *testing.T) {
+	deploymentName := "deployment/nginx-deployment"
+	ns := scheme.Codecs.WithoutConversion()
+	tf := cmdtesting.NewTestFactory().WithNamespace("test")
+	tf.ClientConfigVal = cmdtesting.DefaultClientConfig()
+
+	info, _ := runtime.SerializerInfoForMediaType(ns.SupportedMediaTypes(), runtime.ContentTypeJSON)
+	encoder := ns.EncoderForVersion(info.Serializer, rolloutStatusGroupVersionEncoder)
+	tf.Client = &fake.RESTClient{
+		GroupVersion:         rolloutStatusGroupVersionEncoder,
+		NegotiatedSerializer: ns,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			dep := &appsv1.ReplicaSet{}
+			dep.Name = deploymentName
+			body := io.NopCloser(bytes.NewReader([]byte(runtime.EncodeOrDie(encoder, dep))))
+			return &http.Response{StatusCode: http.StatusNotFound, Header: cmdtesting.DefaultHeader(), Body: body}, nil
+		}),
+	}
+
+	tf.FakeDynamicClient.WatchReactionChain = nil
+	tf.FakeDynamicClient.AddWatchReactor("*", func(action cgtesting.Action) (handled bool, ret watch.Interface, err error) {
```

Comment:
When there is no resource found, `Visit` function and all the clients being used in it will not be triggered and there is no need to `AddWatchReactor`, etc.
This unit test can be simplified.